sources/en.batcave/src/lib.rs

```diff
@@ -132,50 +140,53 @@ impl Source for BatCave {
 		}
 
 		if needs_chapters {
-			let chapter_list: ChapterList = serde_json::from_str(
-				html.select_first(".page__chapters-list > script")
-					.and_then(|x| x.data())
-					.expect("No script data")
-					.strip_prefix("window.__DATA__ = ")
-					.expect("Wrong script format")
-					.strip_suffix(";")
-					.unwrap_or_default(),
-			)
-			.unwrap();
-
-			manga.chapters = Some(
-				chapter_list
-					.chapters
-					.into_iter()
-					.map(|chapter| {
-						let url =
-							format!("{BASE_URL}/reader/{}/{}", chapter_list.news_id, chapter.id);
-
-						let title = chapter
-							.title
-							.strip_prefix(&manga.title)
-							.map(str::trim)
-							.map(String::from)
-							.unwrap_or_else(|| chapter.title);
-
-						let mut chapter_number = None;
-						if let Some(idx) = title.find('#') {
-							chapter_number = title[idx + 1..].parse::<f32>().ok();
-						}
-
-						let date_uploaded = parse_date(&chapter.date, "%-d.%-m.%Y");
+			let script_data = html
+				.select_first(".page__chapters-list > script")
+				.and_then(|x| x.data())
+				.ok_or(error!("No script data"))?;
+
+			let json_str = script_data
+				.strip_prefix("window.__DATA__ = ")
+				.ok_or(error!("Wrong script format"))?
+				.strip_suffix(";")
+				.ok_or(error!("Wrong script format"))?;
```

Comment:
this can be simplified a little bit
```suggestion
				.strip_prefix("window.__DATA__ = ")
				.and_then(|d| d.strip_suffix(";"))
				.ok_or(error!("Wrong script format"))?;
```

Reply:
I was already parsing page lists with a simpler approach, changed here too in 7965d6f